tests/test_barman_wal_archive.py

```diff
@@ -162,11 +162,12 @@ def test_error_barman(self, rpw_mock, tmpdir, capsys):
     def test_connectivity_test_ok(self, popen_mock, capsys):
 
         popen_mock.return_value.communicate.return_value = (b"Good test!", b"")
+        popen_mock.return_value.returncode = 255
```

Comment:
Because the test is testing the successful case it would be clearer to use a `0` here instead of `255`.

Reply:
hmm, I'm a bit confused here.  If I set `popen_mock.return_value.returncode = 0`, then the subsequent `assert exc.value.code == 255` fails (a few lines later).

`walarchive.main(["a.host", "a-server", "--test", "dummy_wal"]` should return 255, since the host and server don't actually exist.  Here's the output when I manually run it:

```
-bash-4.2$ barman-wal-archive a.host a-server --test dummy_wal 
                                                               
-bash-4.2$ echo $?                                             
255                                                            
```

How should I properly construct the test?  Any way not to assign `returncode` to the mock object and see if `walarchive.main()` will cause it to exit with `255`?

Reply:
The test itself is fine but the name of the test tells us it should be testing the "ok" or successful case, so what I was thinking was that we'd have `popen_mock` return a successful exit code and then leave the rest of the test to do what it did originally, i.e. assert that the exit code returned by the command itself is zero.

Alternatively we could continue to use the `255` error code but change the test name to something like `test_connectivity_test_returns_subprocess_output` so that it better matches the behaviour the test is verifying.